experimental/personnames/src/provided_struct.rs

```diff
@@ -0,0 +1,83 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_locid::Locale;
+use litemap::LiteMap;
+use std::mem::{discriminant, Discriminant};
+
+use crate::api::{NameField, PersonName, PersonNamesFormatterError, PreferredOrder};
+
+///
+/// DefaultPersonName, default implementation provided for PersonNameFormatter.
+///
+pub struct DefaultPersonName {
+    person_data: LiteMap<NameField, String>,
```

Comment:
I think the idea is that you can add all these different variants in a locale-friendly way, hopefully always adding the entry with all of the field modifiers disabled, and then we do a best-match lookup. So, if you have an AllCaps entry, and AllCaps is requested, then we find and return it, but otherwise we return the non-capitalized entry _and_ inform the caller that their capitalization preference was not honored, so that they can capitalize it as they want to.

For example, this model allows "McKENNA" to be an explicit all-caps version of "McKenna", whereas a name like "Johnson" could leave the all-caps field blank.

Reply:
gotcha, this makes sense.